bugwarrior/services/deck.py

```diff
@@ -0,0 +1,207 @@
+import datetime
+import logging
+
+import requests
+import typing_extensions
+from dateutil.tz import tzutc
+
+from bugwarrior import config
+from bugwarrior.services import IssueService, Issue, ServiceClient
+
+log = logging.getLogger(__name__)
+
+
+class NextcloudDeckConfig(config.ServiceConfig, prefix='deck'):
+    service: typing_extensions.Literal['deck']
+    base_uri: config.StrippedTrailingSlashUrl
+    username: str
+
+    # can be a password or an app-password
+    password: str
+
+    include_board_ids: config.ConfigList = config.ConfigList([])
+    exclude_board_ids: config.ConfigList = config.ConfigList([])
+
+    import_labels_as_tags: bool = False
+    label_template: str = '{{label}}'
+
+
+# relevant API docs can be found here: https://deck.readthedocs.io/en/latest/API/
+# * Boards will be mapped to projects
+# * Stacks will be mapped to an UDA
+# * Cards will be mapped to tasks
+# * Labels will be mapped to tags
+class NextcloudDeckClient(ServiceClient):
+    def __init__(self, base_uri, username, password):
+        self.base_uri = base_uri
+        self.username = username
+        self.password = password
+        self.api_base_path = 'index.php/apps/deck/api/v1.0'
+
+        self.session = requests.session()
+        self.session.auth = (self.username, self.password)
+        self.session.headers.update({
+            'Accept': 'application/json',
+            'OCS-APIRequest': 'true',
+        })
+
+    # see https://deck.readthedocs.io/en/latest/API/#boards for API docs
+    def get_boards(self):
+        response = self.session.get(f'{self.base_uri}/{self.api_base_path}/boards')
+        return response.json()
+
+    # see https://deck.readthedocs.io/en/latest/API/#stacks for API docs
+    def get_stacks(self, board_id):
+        response = self.session.get(
+            f'{self.base_uri}/{self.api_base_path}/boards/{board_id}/stacks'
+        )
+        return response.json()
+
+    # see https://deck.readthedocs.io/en/latest/API/#comments for API docs
+    def get_comments(self, card_id):
+        response = self.session.get(
+            f'{self.base_uri}/ocs/v2.php/apps/deck/api/v1.0/cards/{card_id}/comments?limit=100&offset=0'
```

Comment:
Looks like you left out `self.api_base_path`. I would combine them in the `__init__` method. (`self.api_base_path = f'{base_uri/index.php/apps/deck/api/v1.0'`) Note that there's also no need to assign any of the `__init__` arguments to `self` if you take this approach as they're only used in the `__init__` method.

Reply:
Oh nevermind, i missed that it is a totally different api path.